capstone/src/main/java/com/google/sps/servlets/CallPerspectiveServlet.java

```diff
@@ -35,15 +37,37 @@
 /** Servlet that returns Perspective scoring. */
 @WebServlet("/call_perspective")
 public class CallPerspectiveServlet extends HttpServlet {
+
   private static final String URL = "https://commentanalyzer.googleapis.com/v1alpha1/comments:analyze?key=API_KEY";
   private static final MediaType JSON = MediaType.get("application/json; charset=utf-8");
 
+  private ApiCaller apiCaller;
+
+  public CallPerspectiveServlet() {
+    super();
+    this.apiCaller = new PerspectiveCaller();
+  }
+
+  public CallPerspectiveServlet(ApiCaller apiCaller) {
+    super();
+    this.apiCaller = apiCaller;
+  }
+
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input
-    String data = request.getReader().readLine();
-    Gson gson = new Gson();		
-    PerspectiveInput info = gson.fromJson(data, PerspectiveInput.class);  
+    PerspectiveInput info = new PerspectiveInput("", "", new String[0]);
+
+    if (request.getReader() != null) {
+      String data = request.getReader().readLine();
+      Gson gson = new Gson();
+      info = gson.fromJson(data, PerspectiveInput.class);
+    }
+
+    if (info == null) {
+      return;
```

Comment:
Do you want to return an HTTP error here?  Maybe 400 (Bad Request, from https://www.restapitutorial.com/httpstatuscodes.html)

Reply:
That makes sense, I'll add that now.